### PR TITLE
Added WS2022 to Makefile build process

### PR DESCRIPTION
**Pull request by nick5616**
**What type of PR is this?**
/kind feature
/kind documentation

**What this PR does / why we need it**:
Enables existing functionality for new OS Windows Server 2022
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
ltsc2022 is the MCR tag for Windows Server 2022
https://hub.docker.com/_/microsoft-windows-nanoserver
There is an identical PR here: https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/807
There is 
**Release note**:
```
none
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,14 +59,14 @@ ALL_OS = linux windows
 ALL_ARCH.linux = amd64 arm64
 ALL_OS_ARCH.linux = $(foreach arch, ${ALL_ARCH.linux}, linux-$(arch))
 ALL_ARCH.windows = amd64
-ALL_OSVERSIONS.windows := 1809 1903 1909 2004
+ALL_OSVERSIONS.windows := 1809 1903 1909 2004, ltsc2022
```
Comment (edreed) [inline Makefile:62]: Values are space delimited, not comma.

Comment (nick5616) [reply Makefile]: Apologies. Mistype

**Comment by edreed** [comment]
The e2e test are failing during the container build step with a timeout pulling package lists from the Debian package repo:

```
#6 [2/4] RUN apt update && apt-mark unhold libcap2
#6 0.109 
#6 0.109 WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
#6 0.109 
#6 0.182 Get:1 http://security.debian.org/debian-security bullseye-security InRelease [44.1 kB]
#6 0.193 Get:2 http://deb.debian.org/debian bullseye InRelease [113 kB]
#6 0.382 Get:3 http://security.debian.org/debian-security bullseye-security/main amd64 Packages [36.5 kB]
#6 30.22 Err:2 http://deb.debian.org/debian bullseye InRelease
#6 30.22   Connection timed out [IP: 199.232.126.132 80]
#6 60.40 Get:4 http://deb.debian.org/debian bullseye-updates InRelease [39.4 kB]
#6 60.60 Get:5 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [2088 B]
#6 60.61 Fetched 122 kB in 60s (2017 B/s)
#6 60.61 Reading package lists...
#6 60.63 Building dependency tree...
#6 60.63 Reading state information...
#6 60.63 1 package can be upgraded. Run 'apt list --upgradable' to see it.
#6 60.63 W: Failed to fetch http://deb.debian.org/debian/dists/bullseye/InRelease  Connection timed out [IP: 199.232.126.132 80]
#6 60.63 W: Some index files failed to download. They have been ignored, or old ones used instead.
#6 60.66 E: Can't select installed nor candidate version from package 'libcap2' as it has neither of them
#6 60.66 E: No packages found
#6 ERROR: process "/bin/sh -c apt update && apt-mark unhold libcap2" did not complete successfully: exit code: 100
```

I just tried an interactive build and it completed without these errors, so perhaps there was an outage? I haven't found any reports of outages for the package repo, though.

I am going to try re-running one of the e2e tests to see if it works today.

**Comment by edreed** [comment]
Interesting. It looks like pull-azuredisk-csi-driver-e2e-migration-windows successfully created and pushed the container image. The re-run of pull-azuredisk-csi-driver-e2e-windows pulled this image and is now running the test case. I'll go ahead and kick off the remainder of the failed e2e tests.

**Review by andyzhangx — APPROVED** [review]
/approve
This is network connection issue in test-infra tests for around 2 days, finally it was recovered.

**Pull request merged**